code/game/machinery/ftl_shieldgen.dm

```diff
@@ -125,16 +137,19 @@
 		icon_state = "[initial(icon_state)]_off"
 
 /obj/machinery/ftl_shieldgen/proc/is_active()
-	return on && plasma_charge >= plasma_charge_max && power_charge >= power_charge_max && istype(loc.loc, /area/shuttle/ftl)
+	return on && plasma_charge >= plasma_charge_min && power_charge >= power_charge_min && istype(loc.loc, /area/shuttle/ftl)
 
 /obj/machinery/ftl_shieldgen/proc/take_hit()
 	spawn(0)
-		drop_physical(1)
 		sleep(1)
-		plasma_charge *= 0.25
-		power_charge *= 0.25
+		power_charge -= min(power_hitloss, power_charge)
 		update_icon()
 		update_physical()
+		var/obj/effect/ftl_shield/hitshield = pick(shield_barrier_objs)
+		hitshield.impact_effect(10)
+		if(is_active())
+			for(var/area/shuttle/ftl/A in world)
```

Comment:
now that the shuttle rewrite is through
`for(var/area in SSshuttle.ftl.shuttle_areas)`